main.tf

```diff
@@ -71,7 +71,7 @@ resource "aws_waf_rule" "this" {
     for_each = (length(each.value.byte_match_tuples) > 0 ? [true] : [])
     content {
       data_id = aws_waf_byte_match_set.this[each.key].id
-      negated = false
+      negated = lookup(each.value, "negated", false)
```

Comment:
If I am not mistaken, the variable type definition of `waf_rules` (https://github.com/Flaconi/terraform-aws-waf-acl-rules/pull/4/files#diff-c9ac8098c5ea9d3e6a9a596ff0c512a4R8) will enforce you to have the `negated` key always set, so using a lookup with a default value would misleading.

Can you confirm or disprove this with a test plan by leaving the `negated` key out of one item of the `waf_rules` variable

Reply:
It is true, it is enforced, in my previous test I had it like this `negated = each.value["negated"]` however though its was more safe to have a default value in case of any wrong value, be default its always false and will not fail if it has other value

Reply:
I ran the plan, it of course fails if the negated its left out:

```
var.default_aws_provider_region
  The AWS region this module is strictly bound to.

  Enter a value: eu-central-1


Error: Unsupported attribute

  on main.tf line 26, in locals:
  26:       negated           = rule.negated

This object does not have an attribute named "negated".
```